From c253f13f534f142172e85092e5cefb8a6b827218 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 06 Nov 2019 12:24:55 +0200
Subject: [PATCH] [SW-0] hl: fixup type from u32 to u64

All SocOnline registers are 32 bit and should not be hold at u64
Also all regmap_* accessors adds HL_CFG_BASE so offset is surly 32 bit

Change-Id: I6ed184b67da9cc43d0609ade4fa3297b877dfc56
---

diff --git a/common/pll.c b/common/pll.c
index 2e9a124..bad1bb3 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -18,7 +18,7 @@
 	cycles_per_usec = hl_pll_get_cpu_cycles_per_usec();
 }
 
-enum hl_pll_bypass_val hl_pll_bypass_get(u64 pll_base, enum hl_pll_div_out out)
+enum hl_pll_bypass_val hl_pll_bypass_get(u32 pll_base, enum hl_pll_div_out out)
 {
 	int offset = out * sizeof(u32);
 
@@ -72,7 +72,7 @@
 static void hl_pll_clk_switch_to_ref(struct hl_pll_cfg *cfg)
 {
 	int sel;
-	u64 reg;
+	u32 reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
@@ -99,7 +99,7 @@
 static void hl_pll_clk_switch_to_pll(struct hl_pll_cfg *cfg)
 {
 	int sel;
-	u64 reg;
+	u32 reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
@@ -138,7 +138,7 @@
 static void hl_pll_div_init(struct hl_pll_cfg *cfg)
 {
 	int sel;
-	u64 div_base;
+	u32 div_base;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		if (cfg->div_sel[sel] == HL_PLL_REF_CLK ||
@@ -195,7 +195,7 @@
 	hl_pll_clk_switch_to_pll(cfg);
 }
 
-u32 hl_pll_get_freq(u64 pll_base, enum hl_pll_div_out div)
+u32 hl_pll_get_freq(u32 pll_base, enum hl_pll_div_out div)
 {
 	u32 nr, nf, od, div_sel, factor, offset;
 	u64 out_freq = SYS_REF_CLOCK;
diff --git a/common/pll.h b/common/pll.h
index 5f77209..3524ae9 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -56,7 +56,7 @@
 
 struct hl_pll_cfg {
 	char	*name;
-	u64	base;
+	u32	base;
 	u32	nr;
 	u32	nf;
 	u32	od;
@@ -66,12 +66,12 @@
 	int	div_wa[HL_PLL_DIV_SIZE];
 };
 
-enum hl_pll_bypass_val hl_pll_bypass_get(u64 pll_base,
+enum hl_pll_bypass_val hl_pll_bypass_get(u32 pll_base,
 					 enum hl_pll_div_out out);
 void hl_pll_init(void);
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg);
 u64 hl_pll_get_cpu_cycles_per_usec(void);
-u32 hl_pll_get_freq(u64 pll_base, enum hl_pll_div_out div);
+u32 hl_pll_get_freq(u32 pll_base, enum hl_pll_div_out div);
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg);
 void hl_pll_udelay(unsigned int usec);
 void hl_pll_wa(void);
diff --git a/common/preboot.c b/common/preboot.c
index 9279d6a..ab5ebcd 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -37,7 +37,7 @@
 	unsigned int next_image, next_image_stable, next_image_modifiable;
 	bool try_new_image = false;
 	char *spl_image = "candidate";
-	u64 upd_pending_reg = get_upd_pending_sts_reg();
+	u32 upd_pending_reg = get_upd_pending_sts_reg();
 
 	modifiable_image_pending = regmap_readl(upd_pending_reg);
 	if (modifiable_image_pending == FLOPS_UPD_PENDING) {
@@ -105,7 +105,7 @@
 	const char *btl_ver_tag = "BTL version ";
 	const char *preboot_ver_str = PREBOOT_VER_STR;
 	char *dst_version_str = (char *)PREBOOT_VER_BASE;
-	u64 offset = (u64)dst_version_str;
+	unsigned long offset = (unsigned long)dst_version_str;
 	u32 btlver = regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
 	int i, btl_ver_sz;
 
diff --git a/common/preboot.h b/common/preboot.h
index 488a535..36b5041 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -24,32 +24,32 @@
 typedef unsigned int u32;
 typedef unsigned long long u64;
 
-static inline u8 regmap_readb(u64 offset)
+static inline u8 regmap_readb(u32 offset)
 {
 	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
 
 	return (u8)(reg & 0xFF);
 }
 
-static inline void regmap_writeb(u8 val, u64 offset)
+static inline void regmap_writeb(u8 val, u32 offset)
 {
 	*((volatile u32 *)(HL_CFG_BASE + offset)) = (u32)val;
 }
 
-static inline u32 regmap_readl(u64 offset)
+static inline u32 regmap_readl(u32 offset)
 {
 	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
 
 	return reg;
 }
 
-static inline void regmap_writel(u32 val, u64 offset)
+static inline void regmap_writel(u32 val, u32 offset)
 {
 	*((volatile u32 *)(HL_CFG_BASE + offset)) = val;
 }
 
-u64 get_cpu_boot_status_reg(void);
-u64 get_upd_pending_sts_reg(void);
+u32 get_cpu_boot_status_reg(void);
+u32 get_upd_pending_sts_reg(void);
 u32 hl_skip_pcie_init(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
diff --git a/common/spi.c b/common/spi.c
index c979432..41a50ba 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -69,7 +69,7 @@
 	regmap_writel(1, mmPSOC_SPI_SSIENR);
 }
 
-static inline int spi_fifo_is_empty(unsigned long level_reg)
+static inline int spi_fifo_is_empty(unsigned int level_reg)
 {
 	return !regmap_readl(level_reg);
 }
diff --git a/common/spi.h b/common/spi.h
index 722c2e9..bba7f04 100644
--- a/common/spi.h
+++ b/common/spi.h
@@ -9,6 +9,6 @@
 #define _HL_SPI_H_
 
 void spi_copy_spl(unsigned int spl_offset);
-u64 get_cs_pad_sel(void);
+u32 get_cs_pad_sel(void);
 
 #endif /* _HL_SPI_H_ */
diff --git a/gaudi/asic.c b/gaudi/asic.c
index c6a4eaf..6f2b5de 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -399,7 +399,7 @@
 {
 	int i;
 	u32 gmask = hl_pll_build_gate_mask(cfg);
-	u64 reg;
+	u32 reg;
 
 	if (gmask != 0xF) {
 		regmap_writel(gmask, cfg->base + HL_PLL_CLK_GATER);
diff --git a/gaudi/common.c b/gaudi/common.c
index b85f8f8..d5bb4af 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -163,17 +163,17 @@
 void preboot_delay(void) {}
 void preboot_putc_virt_uart(const char c) {}
 
-u64 get_cs_pad_sel(void)
+u32 get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
 
-u64 get_cpu_boot_status_reg(void)
+u32 get_cpu_boot_status_reg(void)
 {
 	return mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS;
 }
 
-u64 get_upd_pending_sts_reg(void)
+u32 get_upd_pending_sts_reg(void)
 {
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
diff --git a/goya/common.c b/goya/common.c
index 615826d..4005bd2 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -20,7 +20,7 @@
 
 #define BTL_COUNTER_MAX		BIT(22)
 
-static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
+static int preboot_wait_for_bit(unsigned int reg, unsigned long timeout_ns)
 {
 	unsigned int val;
 	unsigned long count = (timeout_ns * SYS_REF_CLOCK) / 1000000000UL;
@@ -152,17 +152,17 @@
 	regmap_writel(1, mmPCIE_DB_RSV_RSV_1);
 }
 
-u64 get_cs_pad_sel(void)
+u32 get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
 
-u64 get_cpu_boot_status_reg(void)
+u32 get_cpu_boot_status_reg(void)
 {
 	return mmPSOC_GLOBAL_CONF_WARM_REBOOT;
 }
 
-u64 get_upd_pending_sts_reg(void)
+u32 get_upd_pending_sts_reg(void)
 {
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
